packages/dashboard/src/dashboard-actions/dashboardReducer.ts

```diff
@@ -1,25 +1,176 @@
 import { Reducer } from 'redux';
-import { DashboardConfiguration } from '../types';
-import { getMovedDashboardConfiguration } from './move';
+import { DashboardReducerState } from '../types';
 import { DashboardAction } from './actions';
-import { DefaultDashboardConfiguration } from './dashboardState';
+import { move } from './move';
+import { resize } from './resize';
+import { deleteWidgets } from './delete';
+import { paste } from './paste';
+import { undo } from './undo';
+import { redo } from './redo';
+import { dashboardConfig } from './../testing/mocks';
+import { createWidget } from './createWidget';
+import { getSelectedWidgetIds } from '../util/select';
 
-export const dashboardReducer: Reducer<DashboardConfiguration, DashboardAction> = (
-  state: DashboardConfiguration = DefaultDashboardConfiguration,
+const initialState: DashboardReducerState = {
+  dashboardConfiguration: dashboardConfig,
+  intermediateDashboardConfiguration: undefined,
+  selectedWidgetIds: [],
+  numTimesCopyGroupHasBeenPasted: 0,
+  copyGroup: [],
+  stretchToFit: true,
+  width: 1000,
+  cellSize: 30,
+  isActionComplete: true,
+};
+
+export const dashboardReducer: Reducer<DashboardReducerState, DashboardAction> = (
+  state: DashboardReducerState = initialState,
   action: DashboardAction
-): DashboardConfiguration => {
+): DashboardReducerState => {
   switch (action.type) {
-    case 'MOVE': {
-      // return the widget moved to previous position
-      const { position, prevPosition, widgetIds, cellSize } = action.payload;
-      return getMovedDashboardConfiguration({
-        dashboardConfiguration: state,
-        position: position,
-        previousPosition: prevPosition,
-        selectedWidgetIds: widgetIds,
-        cellSize: cellSize,
-      });
-    }
+    case 'MOVE':
+      if (!action.payload.isActionComplete) {
+        if (!state.intermediateDashboardConfiguration) {
+          state.intermediateDashboardConfiguration = state.dashboardConfiguration;
```

Comment:
instead of doing this, on line 40 we can put `dashboardConfiguration: state.intermediateDashboardConfiguration || state.dashboardConfiguration` 

(avoiding mutations, prefer to just have an expression)